packages/cf-component-icon/src/Icon.js

```diff
@@ -1,33 +1,245 @@
+      break;
+    case 'bolt':
+      fontPoint = "'\\F00B'";
+      break;
+    case 'chart':
+      fontPoint = "'\\F00C'";
+      break;
+    case 'drive':
+      fontPoint = "'\\F00D'";
+      break;
+    case 'filter':
+      fontPoint = "'\\F00E'";
+      break;
+    case 'flowchart':
+      fontPoint = "'\\F00F'";
+      break;
+    case 'hamburger':
+      fontPoint = "'\\F010'";
+      break;
+    case 'lock':
+      fontPoint = "'\\F011'";
+      break;
+    case 'list':
+      fontPoint = "'\\F012'";
+      break;
+    case 'file':
+      fontPoint = "'\\F013'";
+      break;
+    case 'shield':
+      fontPoint = "'\\F014'";
+      break;
+    case 'plus':
+      fontPoint = "'\\F015'";
+      break;
+    case 'happy':
+      fontPoint = "'\\F016'";
+      break;
+    case 'sad':
+      fontPoint = "'\\F017'";
+      break;
+    case 'search':
+      fontPoint = "'\\F018'";
+      break;
+    case 'wrench':
+      fontPoint = "'\\F019'";
+      break;
+    case 'remove-sign':
+    case 'remove':
+      fontPoint = "'\\F01A'";
+      break;
+    case 'speech':
+      fontPoint = "'\\F01B'";
+      break;
+    case 'gear':
+      fontPoint = "'\\F01C'";
+      break;
+    case 'help':
+      fontPoint = "'\\F01D'";
+      break;
+    case 'widen':
+      fontPoint = "'\\F022'";
+      break;
+    case 'calendar':
+      fontPoint = "'\\F01E'";
+      break;
+    case 'facebook':
+      fontPoint = "'\\F01F'";
+      break;
+    case 'twitter':
+      fontPoint = "'\\F020'";
+      break;
+    case 'googleplus':
+      fontPoint = "'\\F021'";
+      break;
+    case 'linkedin':
+      fontPoint = "'\\F022'";
+      break;
+    case 'resize-horizontal':
+      fontPoint = "'\\F024'";
+      break;
+    case 'reorder':
+    case 'resize-vertical':
+      fontPoint = "'\\F024'";
+      additionalRules.transform = 'rotate(90deg)';
+      break;
+    case 'upload':
+      fontPoint = "'\\F025'";
+      break;
+    case 'close':
+      fontPoint = "'\\D7'";
+      additionalRules.fontFamily = 'Open Sans, Helvetica, Arial, sans-serif';
+      break;
+  }
+
+  if (border) {
+    additionalRules.border = `1px solid ${color}`;
+    additionalRules.padding = '.2em .25em .15em';
+    additionalRules.borderRadius = '2px';
+  }
 
-    if (this.props.size) {
-      className += ` cf-icon--${this.props.size}`;
+  if (spin) {
+    additionalRules.display = 'inline-block';
+    additionalRules.animation = `${theme.spinAnimationName} 2s infinite linear`;
+  }
+
+  return {
+    color,
+    fontSize,
+    fontStyle: 'normal',
+
+    fontFamily: 'Cloudflare Icons',
+
+    '::before': {
+      ...additionalRules,
+      content: fontPoint
     }
+  };
+}, 'i');
 
-    if (this.props.border) className += ' cf-icon--border';
-    if (this.props.spin) className += ' cf-icon--spin';
-    if (this.props.muted) className += ' cf-icon--muted';
-    if (this.props.white) className += ' cf-icon--white';
+const Label = createComponent(() => ({
+  display: 'none'
```

Comment:
If this is or `a11y` do we need to hide this differently so screen readers can 'see' it?